### PR TITLE
Cleaning of PFEM elements from DEM coupling

### PR DESCRIPTION
**Pull request by AFranci**
PFEM elements for one-fluid analysis are cleaned. 
A new PFEM element is created for handling PFEM-DEM analysis. 
Small changes in PFEM remeshing options.

### CODE REVIEW

**Comment by AFranci** [comment]
Also InvertMatrix is avoided, and lagrangian rotation process is modified (now it is incremental)

File: kratos/includes/constitutive_law.h [hunk 1]
```diff
@@ -821,7 +821,7 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
      * @param rShapeFunctionsValues the shape functions values in the current integration point
      * @param the current ProcessInfo instance
      */
-    KRATOS_DEPRECATED_MESSAGE("Please do not use this method - Use InitializeMaterialResponse instead\"")
+    KRATOS_DEPRECATED_MESSAGE("Please do not use this method\"")
     virtual void InitializeSolutionStep(const Properties& rMaterialProperties,
```
Comment (ipouplana) [inline kratos/includes/constitutive_law.h]: Why did you change this message? Maybe you missed this change?

**Review by ipouplana — COMMENTED** [review]
Good! Please revert the changes in kratos/includes/constitutive_law.h and I'll approve the PR.

File: kratos/includes/constitutive_law.h (hunk 1 above)
Comment (AFranci) [reply kratos/includes/constitutive_law.h]: OK, I have modified them again and put them as in the master. I do not know why the modified after merging with the master branch...

**Review by ipouplana — APPROVED** [review]
